numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -1805,6 +1805,19 @@ gentype_setflags(PyObject *NPY_UNUSED(self), PyObject *NPY_UNUSED(args),
     Py_RETURN_NONE;
 }
 
+/*
+ * Use for concrete np.number subclasses, making them act as if they
+ * were subtyped from e.g. np.signedinteger[object], thus lacking any
+ * free subscription parameters
+ */
+static PyObject *
+numbertype_class_getitem(PyObject *cls, PyObject *args)
+{
+    return PyErr_Format(PyExc_TypeError,
+                        "There are no type variables left in %s",
+                        ((PyTypeObject *)cls)->tp_name);
+}
+
```

Comment:
Mimic the same error used by `types.GenericAlias`: https://github.com/python/cpython/blob/40d2ac92f9a28a486156dafdbb613016bb1f6b98/Objects/genericaliasobject.c#L302-L304